src/bootstrap/tarball.rs

```diff
@@ -262,11 +262,13 @@ impl<'a> Tarball<'a> {
         t!(std::fs::rename(&self.image_dir, &dest));
 
         self.run(|this, cmd| {
+            let distdir = crate::dist::distdir(this.builder);
+            t!(std::fs::create_dir_all(&distdir));
```

Comment:
Can we move this to `fn distdir` rather than special-casing it here? I'm worried we'll see this issue in other commands as well.